### ISSUE TITLE
Idea: Benchmark JNI and native calls

### ISSUE THREAD

**Issue by manticore-projects** [renaissance-benchmarks/renaissance#430]
Greetings.

Accidentally I stumbled over something, which may be interesting.
I tried to encode images to PNG as fast as possible and wrapped native SSE2/AVX optimized encoders into java library. Please see https://github.com/manticore-projects/fpng-java for details.

However, I noticed a massive variance between running my benchmark on GraalVM vs. Open JDK or Jetbrains JDK:

**GRAAL VM**
```text
Benchmark                                           (imageName)  Mode  Cnt     Score    Error  Units
FPNGEBenchmark.encode                               example.png  avgt    3     5.895 ±  0.369  ms/op
FPNGEBenchmark.encode                   looklet-look-scale6.png  avgt    3   242.326 ±  3.215  ms/op
FPNGEncoderBenchmark.encode                         example.png  avgt    3     7.267 ±  4.725  ms/op
FPNGEncoderBenchmark.encode             looklet-look-scale6.png  avgt    3   353.969 ± 14.833  ms/op
```

**OpenJDK**
```text
Benchmark                                           (imageName)  Mode  Cnt     Score    Error  Units
FPNGEBenchmark.encode                               example.png  avgt    3    11.688 ±  0.285  ms/op
FPNGEBenchmark.encode                   looklet-look-scale6.png  avgt    3   530.782 ± 39.565  ms/op
FPNGEncoderBenchmark.encode                         example.png  avgt    3    13.006 ±  0.230  ms/op
FPNGEncoderBenchmark.encode             looklet-look-scale6.png  avgt    3   650.211 ± 38.461  ms/op
```

After some investigation, I found out about the `Graal Compiler`, which  is activated, but can be de-activated (`-XX:-UseJVMCICompiler`, `-XX:-EnableJVMCI`):
```text
FPNGE small image **with** Graal Compiler: 6 ms
FPNGE small image **without** Graal Compiler: 12 ms
    (using  '-XX:-UseJVMCICompiler', '-XX:-EnableJVMCI' switches)
```

Long story short: `Graal VM` with activated `Graal Compiler` seems to provide material advantages when calling JNI/Native Libraries (at least when large memory is transferred). Maybe we want and can add some benchmarks for comparing this specific area.

**Comment by manticore-projects** [comment]
JMH based benchmarks are provided as a sub project here: https://github.com/manticore-projects/fpng-java/tree/main/benchmark

**Comment by manticore-projects** [comment]
Fair enough, thank you for your explanation.
